test/test_sparse.py

```diff
@@ -3276,6 +3277,20 @@ def test_sparse_consistency(self, device, dtype, op):
         assert torch.is_tensor(output)
         self.assertEqual(output.to_dense(), expected)
 
+        inplace_variant = op.get_inplace()
```

Comment:
This is a good extension of this test to handle inplace consistency checking. Instead of querying the promotes_integers_to_float metadata, however, run the operation out-of-place and check if the output can be safely cast to the input's dtype with torch.can_cast:

https://pytorch.org/docs/master/generated/torch.can_cast.html?highlight=can_cast#torch.can_cast

We're going to remove `promotes_integers_to_float` soon in favor of checking for safe casting.

Reply:
@mruberry - Have made the changes and removed the change for promotes_integers_to_float metadata instead checking with torch.can_cast; as per your suggestion